File: lib/models/transaction.info.model.dart
import 'dart:typed_data';

class TransactionInfoModel {
  int? id;
  Uint8List externalTransactionID;
  int amountInSATS;
  int feeInSATS;
  int isSend;
  int transactionTime;
  int feeMode;
  String serverWalletID;
  String serverAccountID;
  String toEmail;
  String toBitcoinAddress;

  TransactionInfoModel({
    required this.id,
    required this.externalTransactionID,
    required this.amountInSATS,
    required this.feeInSATS,
    required this.isSend,
    required this.transactionTime,
    required this.feeMode,
    required this.serverWalletID,
    required this.serverAccountID,
    required this.toEmail,
    required this.toBitcoinAddress,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'externalTransactionID': externalTransactionID,
      'amountInSATS': amountInSATS,
      'feeInSATS': feeInSATS,
      'isSend': isSend,
      'transactionTime': transactionTime,
      'feeMode': feeMode,
      'serverWalletID': serverWalletID,
      'serverAccountID': serverAccountID,
      'toEmail': toEmail,
      'toBitcoinAddress': toBitcoinAddress,
    };
  }

  factory TransactionInfoModel.fromMap(Map<String, dynamic> map) {
    return TransactionInfoModel(
      id: map['id'],
      externalTransactionID: map['externalTransactionID'],
      amountInSATS: map['amountInSATS'],
      feeInSATS: map['feeInSATS'],
      isSend: map['isSend'],
      transactionTime: map['transactionTime'],
      feeMode: map['feeMode'],
      serverWalletID: map['serverWalletID'],
      serverAccountID: map['serverAccountID'],
      toEmail: map['toEmail'],
      toBitcoinAddress: map['toBitcoinAddress'],
    );
  }
}

File: rust/src/proton_wallet/db/model/transaction_info_model.rs
// use rusqlite::{Result, Row};
// use serde::{Deserialize, Serialize};
// use serde_rusqlite::from_row;

// #[derive(Debug, Serialize, Deserialize)]
// pub struct TransactionInfoModel {
//     pub id: Option<u32>,
//     pub external_transaction_id: Vec<u8>,
//     pub amount_in_sats: u32,
//     pub fee_in_sats: u32,
//     pub is_send: u32,
//     pub transaction_time: u32,
//     pub fee_mode: u32,
//     pub server_wallet_id: String,
//     pub server_account_id: String,
//     pub to_email: String,
//     pub to_bitcoin_address: String,
// }

// impl ModelBase for TransactionInfoModel {
//     fn from_row(row: &Row) -> Result<Self> {
//         Ok(from_row::<TransactionInfoModel>(row).unwrap())
//     }
// }
